### ISSUE TITLE
adding support for grouping and sub-grouping

### ISSUE THREAD

**Issue by macaty** [louislam/uptime-kuma#3063]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Uptime-kuma is an amazing project, and after my initial experience using it, I deeply appreciate its beauty. I would love to contribute some great ideas to this project and hope they can be accepted to make kuma even more perfect.

### ✔️ Solution

adding support for grouping and sub-grouping, which would allow users to easily categorize monitoring items and set alert notification recipients by group, or proxies setting

**Comment by JackBailey** [comment]
I'd love this feature too.
More context for maintainers:

Say I have 10 services on a server. One of which being a push.

I want to know when _any_ of my services go down, but if the push monitor is down, the rest would be too. I don't need notifications for every all 10 services, just 1 for the push.

**Comment by CommanderStorm** [comment]
@macaty 
is this one of the PRs, which will be resolved once https://github.com/louislam/uptime-kuma/pull/2693 is merged in the next release?
If yes, could you please cose this issue, as this is one of the duplicates mentioned in above's issue?
If no, could you please rephrase your issue to make it more distinct?

**Comment by CommanderStorm** [comment]
@rwjack
I don't the intention of your comment.
Grouping is in the [current beta](https://github.com/louislam/uptime-kuma/releases/tag/1.22.0-beta.0).

If you'd like an enhancement of said feature, please post a separate issue. Like this, this is not managable.

**Comment by CommanderStorm** [comment]
This issue was partially resolved in https://github.com/louislam/uptime-kuma/pull/2693 and is available since `v1.22.0`.
The remaining issues of this project are tracked in different issues:
- @JackBailey 
  > I want to know when any of my services go down
  
  This is tracked in #1089 and related duplicates
- @macaty 
  > adding support for grouping and sub-grouping, which would allow users to [...] set [...] proxies setting [by group]
 
  This is tracked in #1097

=> closing as resolved
